packages/terra-spacer/src/Spacer.jsx

```diff
@@ -0,0 +1,177 @@
+   */
+  marginTop: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets bottom margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  marginBottom: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets left margin One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  marginLeft: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets right margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  marginRight: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets top padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  paddingTop: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets bottom padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  paddingBottom: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets left padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  paddingLeft: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets right padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  paddingRight: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets the display to be inline-block.
+   */
+  isInlineBlock: PropTypes.bool,
+  /**
+   * Object used to determine at what window size props should be active. e.g { 768: { marginBottom: 'small-2' }}
+   */
+  bps: PropTypes.object,
+};
+
+const defaultProps = {
+  children: null,
+  marginTop: 'none',
+  marginBottom: 'none',
+  marginRight: 'none',
+  marginLeft: 'none',
+  paddingTop: 'none',
+  paddingBottom: 'none',
+  paddingLeft: 'none',
+  paddingRight: 'none',
+  isInlineBlock: false,
+  bps: null,
+};
+
+const Spacer = ({
+  marginTop,
+  marginBottom,
+  marginLeft,
+  marginRight,
+  paddingTop,
+  paddingBottom,
+  paddingLeft,
+  paddingRight,
+  isInlineBlock,
+  children,
+  bps,
+  ...customProps
+ }) => {
+  const SpacerClassNames = cx([
+    'spacer',
+    { 'margin-top-none': marginTop === 'none' },
```

Comment:
I think we can simplify all this by adding a hash like: 
`const spacingSizes  = {`
`none: 'none',`
`'small-2': 'small-minus-2'`,
...
` } `
And then for these classes you would just need to do: 
` { 'margin-top-${spacingSizes[marginTop]} : marginTop }`

Reply:
+3

Reply:
Updated https://github.com/cerner/terra-core/commit/51079e4e1f7a6e3bb384bba084478bb7fcf12335#diff-de1008cdf5bf24f4e07b96189a459cadR85